core/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -586,6 +612,16 @@ protected void writeSnapshotList(List<SnapshotId> snapshots) throws IOException
      * @throws IOException I/O errors
      */
     protected List<SnapshotId> readSnapshotList() throws IOException {
+        try {
+            if (snapshotsBlobContainer.blobExists(SNAPSHOTS_FILE) == false) {
```

Comment:
Regarding URL Repository having blobExists, I suppose you are right, it could just be a HEAD request.  I'm not sure why its not done, but its worth implementing or trying to and see what issues arise (in a separate PR).